source/frontend/editors/RarcEditor.cpp

```diff
@@ -0,0 +1,1035 @@
+    {"bpk", s_animation_info},     {"brk", s_animation_info},
+    {"btk", s_animation_info},     {"btp", s_animation_info},
+    {"bva", s_animation_info},     {"arc", s_archive_info},
+    {"carc", s_archive_info},      {"u8", s_archive_info},
+    {"col", s_collision_info},     {"kcl", s_collision_info},
+    {"szs", s_compressed_info},    {"bdof", s_environment_info},
+    {"bfg", s_environment_info},   {"blight", s_environment_info},
+    {"blmap", s_environment_info}, {"pblm", s_environment_info},
+    {"bin", s_environment_info},   {"ral", s_environment_info},
+    {"ymp", s_environment_info},   {"bfn", s_font_info},
+    {"pad", s_input_info},         {"blo", s_layout_info},
+    {"brlyt", s_layout_info},      {"brlan", s_layout_info},
+    {"brctr", s_layout_info},      {"bmt", s_material_info},
+    {"bmg", s_message_info},       {"bdl", s_model_info},
+    {"bmd", s_model_info},         {"brres", s_model_info},
+    {"dae", s_model_info},         {"fbx", s_model_info},
+    {"3ds", s_model_info},         {"thp", s_movie_info},
+    {"jpa", s_particle_info},      {"sb", s_script_info},
+    {"bas", s_sound_info},         {"bms", s_sound_info},
+    {"com", s_sound_info},         {"scom", s_sound_info},
+    {"aw", s_sound_info},          {"afc", s_sound_info},
+    {"ws", s_sound_info},          {"bnk", s_sound_info},
+    {"bld", s_sound_info},         {"bst", s_sound_info},
+    {"bti", s_texture_info},       {"bmp", s_texture_info},
+    {"kmp", s_track_info},
+};
+
+void RarcEditorPropertyGrid::Draw(ResourceArchive& rarc, RarcEditor* editor) {
+  RSL_DEFER(editor->ApplyUpdates());
+
+  constexpr auto dir_flags = ImGuiTreeNodeFlags_OpenOnArrow |
+                             ImGuiTreeNodeFlags_OpenOnDoubleClick |
+                             ImGuiTreeNodeFlags_SpanFullWidth;
+
+  constexpr auto file_flags =
+      ImGuiTreeNodeFlags_Leaf | ImGuiTreeNodeFlags_SpanFullWidth;
+
+  constexpr auto root_flags = dir_flags | ImGuiTreeNodeFlags_DefaultOpen;
+
+  if (m_context_modal_state == ModalState::M_CLOSED && m_operation_modal_state == ModalState::M_CLOSED)
+    m_focused_node = std::nullopt;
+
+  ImGui::Text("Search");
+  ImGui::SameLine();
+  m_model_filter.Draw();
+
+  if (ImGui::BeginTable("TABLE", 3,
+                        ImGuiTableFlags_RowBg |
+                            ImGuiTableFlags_NoSavedSettings)) {
+    RSL_DEFER(ImGui::EndTable());
+
+    ImGui::TableSetupScrollFreeze(3, 1);
+    ImGui::TableSetupColumn("Node Tree");
+    ImGui::TableSetupColumn("Flags", ImGuiTableColumnFlags_WidthFixed, 240.0f);
+    ImGui::TableSetupColumn("Size", ImGuiTableColumnFlags_WidthFixed, 50.0f);
+    ImGui::TableHeadersRow();
+
+    std::vector<std::size_t> walk_stack;
+
+    std::size_t row = 0;
+    for (auto node = rarc.nodes.begin(); node != rarc.nodes.end(); row++) {
+      // We don't render . and ..
+      if (node->is_special_path()) {
+        node++;
+        continue;
+      }
+
```

Comment:
Do we need this now?

Reply:
Well the special paths are in the RARC node list and this keeps us from rendering rows for them.

Reply:
Could we have a library function to remove special dirs, so the UI doesn't have to see them? And another to add them back

Reply:
I suppose I could since the editor is the only thing using the RARC.

Reply:
Do special directories even need to be exposed to library users?

Reply:
If I've already created methods to remove and insert them wholly, then no. But also, I believe you had them exposed in the U8 code so unless I'm mistaken there is that to consider.